### PR TITLE
feat(upgrade): auto upgrade

### PR DESCRIPTION
**Pull request by JanDeDobbeleer**
- **feat(upgrade): use built-in upgrade by default**
- **feat(upgrade): auto upgrade**
- **chore(go): update dependencies**

### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md].
- [x] The commit message follows the [conventional commits][cc] guidelines.
- [x] Tests for the changes have been added (for bug fixes / features).
- [x] Docs have been added/updated (for bug fixes / features).

### Description

This PR introduces the `auto_upgrade` configuration parameter which will trigger the
upgrade using the same interval as the upgrade notice on shell startup.

Currently a darkly launched feature so it can be experimented with.

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -74,7 +74,7 @@
         "prompt",
         "init",
         "cmd",
-        "--print"
+        "--config=/Users/jan/.posh.omp.jsonc"
```
Comment (JanDeDobbeleer) [inline .vscode/launch.json]: needs to be removed

**Comment by kimsey0** [comment]
@JanDeDobbeleer, thanks for the new auto-upgrade feature. I don't know where, if anywhere, you take feedback on it, but it would be great to remove or at least allow disabling the `✅ no new version available` notice in every new terminal window. It doesn't really convey any information, so it's just noise to me.

https://github.com/JanDeDobbeleer/oh-my-posh/blob/170e5ac9e5b12d918240add12078a0fadd71330b/src/cli/upgrade.go#L42

**Comment by JanDeDobbeleer** [comment]
@kimsey0 that's solved by adding `disable_notice: true`

**Comment by kimsey0** [comment]
But does that still show the actual upgrade message?

**Comment by JanDeDobbeleer** [comment]
Yes, when it upgrades it will still show that.

**Comment by kimsey0** [comment]
OK. I think this interaction sounds a bit confusing when spelled out ("`disable_notice` disables the notice that a new upgrade is available when `auto_upgrade` is disabled, and the notice that no new upgrade is available when `auto_upgrade` is enabled"), but would it still make sense to document, as in #5175?

**Pull request merged**